src/IIIFPresentation/API/Features/Manifest/ManifestService.cs

```diff
@@ -207,11 +223,29 @@ await canvasPaintingResolver.InsertCanvasPaintings(request.CustomerId, request.P
         };
         
         await dbContext.AddAsync(dbManifest, cancellationToken);
+        
+        if (request.PresentationManifest.PaintedResources.HasAsset())
```

Comment:
`request.PresentationManifest.PaintedResources.HasAsset()` is called twice in this method, is it worth storing as a variable to avoid potentially enumerating the `paintedResources` multiple times? Alternatively store all assets in a list and check for the length of that, which could save enumerating again for `request.PresentationManifest.PaintedResources!.Select(p => p.Asset).ToList()` on line 232